django/chat/responses.py

```diff
@@ -211,44 +229,47 @@ def summarize_response(chat, response_message):
             content_type="text/event-stream",
         )
     elif user_message.text == "":
-        summary = _("No text to summarize.")
+        error_str = _("No text to summarize.")
     else:
+        # Text input is a URL or plain text
         url_validator = URLValidator()
         try:
             url_validator(user_message.text)
             text_to_summarize = url_to_text(user_message.text)
+            # Check if response text is too short (most likely a website blocking Otto)
+            if len(text_to_summarize.split()) < 35:
+                error_str = _(
+                    "Couldn't retrieve the webpage. The site might block bots. Try copy & pasting the webpage here."
```

Comment:
This logic was poorly nested before so it would sometimes occur even when user hadn't entered a URL.